src/main/java/io/cdap/plugin/gcp/gcs/actions/GCSBucketDelete.java

```diff
@@ -119,13 +128,54 @@ public void run(ActionContext context) throws Exception {
       if (fs.exists(gcsPath)) {
         try {
           fs.delete(gcsPath, true);
+          deleteCount++;
         } catch (IOException e) {
           LOG.warn(
             String.format("Failed to delete path '%s'", gcsPath)
           );
         }
       }
     }
+
+    for (GCSPath gcsPath : gcsPathsWild) {
+      String regex = ("\\Q" + gcsPath.getUri().toString() + "\\E").replace("*", "\\E[^/]*\\Q") + "(/.*)?";
+      try {
+        fs = new Path(gcsPathsWild.get(0).getUri()).getFileSystem(configuration);
```

Comment:
still using `gcsPathsWild.get(0)` ? then we can move it out of the iteration ?

Reply:
Setting another Jira to solve the delete from multiple bucket: https://cdap.atlassian.net/browse/PLUGIN-1291.